subsys/ipc/ipc_service/backends/Kconfig

```diff
@@ -1,46 +1,37 @@
 # Copyright (c) 2021 Nordic Semiconductor (ASA)
 # SPDX-License-Identifier: Apache-2.0
 
-choice IPC_SERVICE_BACKEND
-	prompt "IPC service backend"
+DT_COMPAT_ZEPHYR_IPC_OPENAMP_STATIC_VRINGS := zephyr,ipc-openamp-static-vrings
+
+config IPC_SERVICE_REG_BACKEND_PRIORITY
+	int "Initialization priority of modules registering IPC backend"
+	default 46
+	help
+	  The backend must be registered before the endpoint register.
 
 config IPC_SERVICE_BACKEND_RPMSG
 	bool "OpenAMP RPMSG backend with static VRINGs"
 	depends on MBOX
+	default $(dt_compat_enabled,$(DT_COMPAT_ZEPHYR_IPC_OPENAMP_STATIC_VRINGS))
 	select IPC_SERVICE_RPMSG
 	select IPC_SERVICE_STATIC_VRINGS
 	select OPENAMP
 
-config IPC_SERVICE_BACKEND_ZTEST
-	depends on ZTEST
-	bool "IPC service backend test"
-
-endchoice
-
-config IPC_SERVICE_NUM_ENDPOINTS_PER_INSTANCE
-	int "Max number of registered endpoints per instance"
-	default 2
-	help
-	  Maximal number of endpoints that can be registered for one instance.
-
-config IPC_SERVICE_REG_BACKEND_PRIORITY
-	int "Initialization priority of modules registering IPC backend"
-	default 46
-	help
-	  The backend must be registered before the endpoint register.
-
 config IPC_SERVICE_RPMSG
 	bool "RPMsg support library"
+	depends on IPC_SERVICE_BACKEND_RPMSG
```

Comment:
This doesn't actually depends on that backend. The backend is selecting this library, but the library can be selected by multiple backends. So, this is not needed.